app/src/main/res/layout/item_episode.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    style="@style/ParentWidthWrapHeight"
+    android:layout_margin="4dp"
+    app:cardCornerRadius="4dp"
+    app:cardElevation="6dp"
+    app:contentPadding="4dp">
+
+    <androidx.appcompat.widget.LinearLayoutCompat
```

Comment:
Czy orientujesz się może czy nadal takie wpisy są aktualne ? 
https://android.jlelse.eu/constraint-layout-performance-870e5f238100
Chodzi o to że często starsze widgety jak LinearLayut do umieszczenia poziomo elementów zachowują się wydajniej niż umieszczanie ich w Constraint Layout, mowa o tym żeby użyć czegoś najprostszego bo to będzie najszbyciej działać. Widziałem że niektórzy używali jednak Constraint Layoutu w kursie i stąd moje pytanie: jak u ciebie wygląda to w praktyce ? Bo w przykładach google/a np. to przykładowe repozytorium z roślinkami "sunflower" też zawiera użycia frameLAyoutu,LinearLayoutu itd.

Reply:
Niestety nie wiem jak wygląda wydajność constraint layoutu w najnowszej wersji względem tych starszych kontenerów na widgety. W projektowaniu widoków staram się być pragmatyczny. Jeśli coś da się zrobić prosto przy użyciu `LinearLayout` albo `FrameLayout` to z nich korzystam. Gdy trzeba zrobić bardziej zaawansowany widok, gdzie rozmieszczenie poszczególnych elementów jest nieszablonowe wtedy sięgam po `ConstraintLayout`. Staram się trzymać zasady, że gdy muszę zagnieździć te podstawowe kontenery, żeby uzyskać efekt jaki mnie interesuje, to wtedy używam `ConstraintLaoyut`. Gdy da się zrobić płaską strukturę tymi prostymi kontenerami, to wtedy z nich korzystam.

Reply:
Plus trzeba mieć na uwadze, że nowa wersja `ConstraintLaoyutu` umożliwia korzystanie z `MotionLayout` więc jak ktoś chce robić jakieś fajne animacje przy użyciu tego narzędzia, to raczej nie ma za dużego wyboru :)